### PR TITLE
[17.01] explicit sniffing of FastqGz/FastqBz2 types

### PR DESCRIPTION
**Pull request by jvolkening**
In my testing of the new compressed FASTQ datatypes, bzip2-compressed files were being recognized as "fastq.gz", which led to silent failure of the conversion tool when used as tool input. It seems that this is because the sniffer used for both is from the FastqBase parent class, which returns true for any of the derived classes (including these two), so the recognition is simply based on the order in which the calls to `Binary.register_sniffable_binary_format()` are made in `lib/galaxy/datatypes/sequence.py`.

The attached PR addressses this issue by defining explicit `sniff()` methods for each derived class, which check the compression type before calling the parental method.

It also seems that the quality-encoding-specific versions should not be registered as sniff-able, as they will always (as currently defined) be recognized as their parental class.

### CODE REVIEW

File: lib/galaxy/datatypes/sequence.py [hunk 1]
```diff
@@ -721,6 +721,11 @@ class FastqGz ( BaseFastq, Binary ):
     edam_format = "format_1930"
     file_ext = "fastq.gz"
 
+    def sniff( self, filename ):
+        """Determines whether the file is in gzip-compressed FASTQ format"""
+        if not is_gzip(filename):
+            return False
+        return BaseFastq.sniff( self, filename )
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/sequence.py:728]: Can you add one more blank line here, please?

**Review by mvdbeek — APPROVED** [review]
Thanks a lot @jvolkening 
If there are no objections I'm gonna merge in the evening.

**Comment by nsoranzo** [comment]
This PR is going to conflict with @jmchilton's #3510.

**Review by martenson — APPROVED** [review]

**Comment by martenson** [comment]
Looks good. Thank you very much for the contribution @jvolkening !

**Pull request merged**